platform/state/platform_state.proto

```diff
@@ -0,0 +1,391 @@
+     * <p>
+     * This SHALL be updated whenever emergency recovery is performed.
+     */
+    bytes epoch_hash = 7;
+
+    /**
+     * A SHA-384 hash that will be the next "epoch" hash.
+     * <p>
+     * This SHALL be used to update the epoch hash at the next round boundary.<br/>
+     * This SHALL NOT be part of the hash and SHALL NOT be serialized.
+     */
+    bytes next_epoch_hash = 8;
+
+    /**
+     * A number of non-ancient rounds.
+     * <p>
+     * This SHALL be the count of rounds considered non-ancient.
+     */
+    uint32 rounds_non_ancient = 9;
+
+    /**
+     * A snapshot of the consensus state at the end of the round.
+     * <p>
+     * This SHALL be used for restart/reconnect.
+     */
+    ConsensusSnapshot consensus_snapshot = 10;
+
+    /**
+     * A timestamp for the next scheduled time when a freeze will start.
+     * <p>
+     * If a freeze is not scheduled, this SHALL NOT be set.
+     */
+    proto.Timestamp freeze_time = 11;
+
+    /**
+     * A timestamp for the last time a freeze was performed.<br/>
+     * If not set, there has never been a freeze.
+     */
+    proto.Timestamp last_frozen_time = 12;
+
+    /**
+     * A consensus node software version.<br/>
+     * The software version that enabled birth round mode.
+     * <p>
+     * This SHALL be unset if birth round migration has not yet happened.<br/>
+     * Otherwise, this SHALL be the _first_ software version that enabled birth
+     * round mode.
+     */
+    SoftwareVersion first_version_in_birth_round_mode = 13;
+
+    /**
+     * A consensus round.<br/>
+     * The last round before the birth round mode was enabled.
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 last_round_before_birth_round_mode = 14;
+
+    /**
+     * A consensus generation.<br/>
+     * The lowest judge generation before birth round mode was enabled.
+     * <p>
+     * This SHALL be -1 if birth round mode has not yet been enabled.
+     */
+    int64 lowest_judge_generation_before_birth_round_mode = 15;
+}
+
+
+/**
+ * A consensus node software version.
+ *
+ * This message records version information for configuration, the Hedera
+ * API, and the "Services" subsystems.
+ */
+message SoftwareVersion {
+    /**
+     * A single numeric version.<br/>
+     * This is the current configuration version read on node startup.
+     */
+    uint32 config_version = 1;
+
+    /**
+     * A semantic version entry.<br/>
+     * This is the version of the HAPI module (Hedera API).
+     */
+    proto.SemanticVersion hapi_version = 2;
+
+    /**
+     * A semantic version entry.<br/>
+     * This is the version of the services module.
+     */
+    proto.SemanticVersion services_version = 3;
+}
+
+
+/**
+ * A consensus snapshot.<br/>
+ * This is a snapshot of the consensus state for a particular round.
+ *
+ * This message SHALL record consensus data necessary for restart
+ * and reconnect.
+ */
+message ConsensusSnapshot {
```

Comment:
Should different protobuffs be split into their own files?

Reply:
`consensus_service.proto` suggests that we may have multiple messages in one file. Also, @jsync-swirlds didn't object to it, so I think it should be okay.

Reply:
Organization of messages in files is relatively low concern.  It's minimal impact to move things around later if we decide to.

That said, closely related messages are reasonable in a single file, but unrelated messages really should be in different files generally.